### ISSUE TITLE
Cli inline config or configuration in `package.json`

### ISSUE THREAD

**Issue by JounQin** [dotansimha/graphql-code-generator#5140]
**Is your feature request related to a problem? Please describe.**

Sometimes, the configuration is just so small and I'd prefer to add it into `package.json` or even use cli inline config instead:

```yaml
overwrite: true
schema: "schema.graphql"
generates:
  schema.d.ts:
    plugins:
      - "typescript"
```

**Describe the solution you'd like**

As title, config inline or in `package.json`.

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

**Comment by dotansimha** [comment]
Hi @JounQin 
We are using `cosmiconfig` to manage configuration file. It support many forms of loading configurations.

You can use codegen config as JSON, and put it under your `package.json` and it will work the same:

```json
"codegen": {
  "schema": "schema.graphql",
  // ...
}
```

**Comment by JounQin** [comment]
@dotansimha Thanks for your reply, it's not well documented.

**Comment by dotansimha** [comment]
@JounQin you are right, I'll update the website. Thanks!

**Comment by JounQin** [comment]
@dotansimha Sorry but I tried to add `codegen` in `package.json` but failed:

```log
yarn graphql-codegenyarn run v1.22.10
$ /Users/JounQin/Workspaces/GitHub/graphql-schema/node_modules/.bin/graphql-codegen
Unable to find Codegen config file!
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by JounQin** [comment]
The `cosmiconfig` search places are not by default, which may cause confusion.

https://github.com/dotansimha/graphql-code-generator/blob/master/packages/graphql-codegen-cli/src/config.ts#L21-L29

**Comment by dotansimha** [comment]
Yeah, you are right. Just noticed it. 
I created a PR here: https://github.com/dotansimha/graphql-code-generator/pull/5144 
Can you please try the alpha version from that PR?

**Comment by JounQin** [comment]
@dotansimha The alpha version works.

**Comment by dotansimha** [comment]
Perfect, thank you for helping me to test it. I will merge and release soon :)

**Comment by dotansimha** [comment]
Fixed in `@graphql-codegen/cli@1.19.4`. Thanks @JounQin !
